src/server/tiered_storage.cc

```diff
@@ -451,13 +451,7 @@ error_code TieredStorage::ScheduleOffload(DbIndex db_index, PrimeIterator it, st
   }
 
   if (blob_len > kMaxSmallBin) {
-    auto [schedule, res_it] = CanScheduleOffload(db_index, it, key);
-    if (schedule) {
-      WriteSingle(db_index, res_it, blob_len);
-    } else {
-      VLOG(2) << "Skip WriteSingle for: " << key;
-    }
-    return error_code{};
+    return true;
```

Comment:
why do you exit?

Reply:
This function just does some setup and returns if we can schedule offloading. When writing big blobs we always schedule offload, while writing small blobs we aggregate in pending_entries and only if pending_entries is max size we schedule offloading. 
So below is the preparation for small blobs where we add to pending_entries and we will return true (will result in scheduleing for offload from the caller) if we aggregated max entries in the bin